File: 2022/rust/src/main.rs
use std::fs;

use aoc::{from_input, most_calories, top3_calories};

fn read(day: &str) -> String {
    return fs::read_to_string(format!("../inputs/{}", day))
        .expect(&format!("Could not open {} data", day));
}

/*
   cargo run
*/
pub fn main() {
    let input = read("day01");
    let calories = from_input(&input);

    let answer1 = most_calories(calories.clone());
    println!("Answer 1: {}", answer1);

    let answer2 = top3_calories(calories);
    println!("Answer 2: {}", answer2);
}

File: 2022/go/main.go
package main

import (
	"aoc/day01"
	"fmt"
	"os"
)

func main() {
	file, err := os.ReadFile("../_inputs/day01")
	if err != nil {
		panic(err)
	}
	fmt.Println("Day 01")
	fmt.Println("======")
	fmt.Printf("Answer 1: %d\n", day01.MostCalories(string(file)))
	fmt.Printf("Answer 2: %d\n", day01.Top3Calories(string(file)))
	fmt.Println("\n---\n")
}
